src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -257,11 +261,15 @@ public TopicPartitionChannel(
     /* Schematization related properties */
     this.enableSchematization =
         this.recordService.setAndGetEnableSchematizationFromConfig(sfConnectorConfig);
-    this.enableSchemaEvolution =
-        this.enableSchematization
-            && this.conn != null
-            && this.conn.hasSchemaEvolutionPermission(
-                tableName, sfConnectorConfig.get(SNOWFLAKE_ROLE));
+
+    if (hasSchemaEvolutionPermission == null) {
+      hasSchemaEvolutionPermission =
+          this.conn != null
+              && this.conn.hasSchemaEvolutionPermission(
+                  tableName, sfConnectorConfig.get(SNOWFLAKE_ROLE));
```

Comment:
nit: these three condition checks could be combined?

Reply:
The boolean expression might looks like this:
```
hasSchemaEvolutionPermission = 
    (hasSchemaEvolutionPermission != null && hasSchemaEvolutionPermission) ||
    (hasSchemaEvolutionPermission == null &&
     this.conn != null && 
     this.conn.hasSchemaEvolutionPermission(tableName, sfConnectorConfig.get(SNOWFLAKE_ROLE))
    );
```
I think it's quit complicate, should I use ternary instead?

Reply:
IMO it could be combined but it is doesn't read well. current way @sfc-gh-alhuang has is more readable.

Reply:
ok, I think the way we're doing this is wried, we should pass a boolean instead of Boolean. It's unexpected that the map doesn't have the table. Then this check could be removed completely, wdyt?

Reply:
@sfc-gh-tzhang make sense. Removed.